Errors on receving sigterm

Kwuray opened the pull request:
### What does this PR do?

This PR fix the bug where error are thrown on SIGTERM signal.


### Motivation

[Issue #10913](https://github.com/traefik/traefik/issues/10913)


### More


### Additional Notes

Maybe it should be a good idea to add a test, to ensure that no error are throwns when entrypoints are gracefully stopped. But the error is not returned, just print via the Logger, so I don't how to do that.

nmengin commented:
Hello @Kwuray,

Thank you for your contribution. We have labeled it as "need-review" to check.

> Maybe it should be a good idea to add a test

I agree, could you add this test?

Thanks in advance.

Kwuray commented:
Hello @nmengin,

I added the test, but I'm not sure I did it the right way.

It's difficult to test that no error are printed into stderr, so I added an error variable into the httpServer struct so we can test it. Not 100% convinced.

I would like to have your opinion on this. Thanks!

nmengin commented:
Hey @Kwuray.

Sorry for the delay.
Your PR is in still our (maintainers') backlog.

> I would like to have your opinion on this.

We are focused elsewhere, and even if we have planned to move forward soon, I have no ETA to provide yet.
We'll keep you updated as soon as we move forward on it.

kevinpollet commented:
Hello @Kwuray and thanks for your contribution,

We updated the pull request by reinstating the HTTPS server shutdown, which was necessary. We also introduced a listener wrapper to ensure the listener closes only once. Finally, we refactored your work slightly and kept the idea of the shutdown boolean.

juliens reviewed (approved):
LGTM

kevinpollet reviewed (approved):
Thanks 👍
